src/utils.jl

```diff
@@ -50,19 +48,21 @@ _adddiag(a::Union{Matrix, SparseMatrixCSC}, v::Real) = _adddiag!(copy(a), v)
 _adddiag(a::Union{Matrix, SparseMatrixCSC}, v::Vector, c::Real) = _adddiag!(copy(a), v, c)
 _adddiag(a::Union{Matrix, SparseMatrixCSC}, v::Vector{T}) where {T<:Real} = _adddiag!(copy(a), v, one(T))
 
+
 function wsumsq(w::AbstractVector, a::AbstractVector)
-    @check_argdims(length(a) == length(w))
+    @check_argdims(eachindex(a) == eachindex(w))
     s = zero(promote_type(eltype(w), eltype(a)))
-    for i = 1:length(a)
+    for i in eachindex(w)
         @inbounds s += abs2(a[i]) * w[i]
     end
     return s
 end
 
 function invwsumsq(w::AbstractVector, a::AbstractVector)
-    @check_argdims(length(a) == length(w))
+    @check_argdims(eachindex(a) == eachindex(w))
```

Comment:
Same as above.

Reply:
This is safe. This makes sure the indexes *are* in-fact the same
This is what I think this PR should do everywhere.
I missed it above.

Making sure that things with different indexes work the same as before would require much more extensive testing.
And i think can be left for another PR.